gcloud-java-logging/src/main/java/com/google/gcloud/logging/spi/v2/ConfigServiceV2Api.java

```diff
@@ -0,0 +1,533 @@
+   * same name that is specified in `sinkName`.  If the sink does not
+   * exist, it is created.
+   * @param sink The updated sink, whose name must be the same as the sink
+   * identifier in `sinkName`.  If `sinkName` does not exist, then
+   * this method creates a new sink.
+   */
+  public LogSink updateSink(String sinkName, LogSink sink) {
+    UpdateSinkRequest request =
+        UpdateSinkRequest.newBuilder()
+        .setSinkName(sinkName)
+        .setSink(sink)
+        .build();
+
+    return updateSink(request);
+  }
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
+  /**
+   * Creates or updates a sink.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   *
+   * @param request The request object containing all of the parameters for the API call.
+   */
+  public LogSink updateSink(UpdateSinkRequest request) {
+    return updateSinkCallable().call(request);
+  }
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
+  /**
+   * Creates or updates a sink.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   */
+  public ApiCallable<UpdateSinkRequest, LogSink> updateSinkCallable() {
+    return ApiUtils.prepareIdempotentCallable(UPDATE_SINK, settings).bind(channel);
+  }
+
+  // ----- deleteSink -----
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
+  /**
+   * Deletes a sink.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   *
+   * @param sinkName The resource name of the sink to delete.
+   * Example: `"projects/my-project-id/sinks/my-sink-id"`.
+   */
+  public void deleteSink(String sinkName) {
+    DeleteSinkRequest request =
+        DeleteSinkRequest.newBuilder()
+        .setSinkName(sinkName)
+        .build();
+
+    deleteSink(request);
+  }
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
+  /**
+   * Deletes a sink.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   *
+   * @param request The request object containing all of the parameters for the API call.
+   */
+  public void deleteSink(DeleteSinkRequest request) {
+    deleteSinkCallable().call(request);
+  }
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
+  /**
+   * Deletes a sink.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   */
+  public ApiCallable<DeleteSinkRequest, Empty> deleteSinkCallable() {
+    return ApiUtils.prepareIdempotentCallable(DELETE_SINK, settings).bind(channel);
+  }
+
+
+  // ========
+  // Cleanup
+  // ========
+
+  /**
+   * Initiates an orderly shutdown in which preexisting calls continue but new calls are immediately
+   * cancelled.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   */
+  @Override
+  public void close() {
+    // Manually-added shutdown code
```

Comment:
Ok, your call, but let me explain my suggestion (at least for the record).

So far it looked to me that your general approach for modifying the class directly by a user is for adding new functionality or documentation rather than modifying  the methods themselves (this is also what you mention in the "EDITING INSTRUCTIONS").

The other approach would be to override, which is also indirectly documented.
 
The "close" method seems to get a "special" treatment (which violates the general editing instruction). I am not sure if that is worth it  (and if so, better documentation would be useful). If not worth it, then one could just achieve
the same by overriding and delegating (between "before" and "after" logic).
If "close" was critical and you would not want people to "mess it up", then
you could make it final and provide the 2 before and after methods to be
overridden.